crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__empty_lines.py.snap

```diff
@@ -163,11 +163,11 @@ def f():
             }:
                 return NO
 
-
 ###############################################################################
```

Comment:
The difference in this file is now just that we aren't adding the extra empty line before this comment (which trails a function).

Reply:
This is an interesting example and is somewhat different from the example that we discussed

```python
class Test: ...
# bar

x =1
```

Because the comment is separated by one line from the class. I can see why black adds two empty because the comment is clearly separate from the preceding statement (not directly adjacent). But implementing this is probably even more complicated than what you had initially.

Reply:
Ideal solution: Only add newlines before comments with at least one empty line.
Okay solution: Reverting to what you had.

Reply:
Alright, I'm reverting to what I had, but I'm going to look into a followup PR to move the comment handling into `suite.rs` and also enable the behavior we discussed here.